src/main/java/com/spaceuptech/api/core/utils/Transport.java

```diff
@@ -0,0 +1,129 @@
+        if (dbType != null) metaBuilder.setDbType(dbType);
+        if (token != null) metaBuilder.setToken(token);
+
+        return metaBuilder.build();
+    }
+
+    private static StreamObserver<Response> makeStreamObserver(Utils.ResponseListener listener, ManagedChannel channel) {
+        return new StreamObserver<Response>() {
+            @Override
+            public void onNext(Response value) {
+
+                System.out.println(value);
+
+//                if (value.getStatus() == 200)
+//                    listener.onResponse(value.getStatus(), new com.spaceuptech.api.core.utils.Response(new JsonParser().parse(value.getResult().toString()).getAsJsonObject()));
+//                else
+//                    listener.onResponse(value.getStatus(), new com.spaceuptech.api.core.utils.Response(new JsonParser().parse(value.getError()).getAsJsonObject()));
+
+                try {
+                    channel.shutdown().awaitTermination(1, TimeUnit.SECONDS);
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
+                }
+            }
+
+            @Override
+            public void onError(Throwable t) {
+                listener.onError(new Exception(t));
+            }
+
+            @Override
+            public void onCompleted() {}
+        };
+    }
+
+    public static void create(String host, int port, Object doc, String operation, Meta meta, Utils.ResponseListener listener) {
+
+        final ManagedChannel channel = ManagedChannelBuilder.forAddress(host, port).usePlaintext().build();
+        final SpaceCloudGrpc.SpaceCloudStub stub = SpaceCloudGrpc.newStub(channel);
+
+        ByteArrayOutputStream bos = new ByteArrayOutputStream();
+        ObjectOutput out;
+        try {
+            out = new ObjectOutputStream(bos);
+            out.writeObject(doc);
+            out.flush();
+            byte[] bytes = bos.toByteArray();
+
+            CreateRequest createRequest = CreateRequest.newBuilder()
+                    .setDocument(ByteString.copyFrom(bytes))
+                    .setOperation(operation)
+                    .setMeta(meta).build();
+
+            stub.create(createRequest, makeStreamObserver(listener, channel));
+        } catch (IOException e) {
+            e.printStackTrace();
+        } finally {
+            try {
+                bos.close();
+            } catch (IOException ex) {
+                // ignore close exception
+            }
+        }
+    }
+
+    public static void read(String host, int port, Object find, String operation, ReadOptions options, Meta meta, Utils.ResponseListener listener) {
+
+        final ManagedChannel channel = ManagedChannelBuilder.forAddress(host, port).usePlaintext().build();
+        final SpaceCloudGrpc.SpaceCloudStub stub = SpaceCloudGrpc.newStub(channel);
+
+        ReadRequest readRequest = ReadRequest.newBuilder()
+                .setFind((ByteString) find)
```

Comment:
Don't do this. Use gson.toJson or something. You need to find the JSON equivalent and not the Java part

Reply:
Done